internal/controller/application_controller.go

```diff
@@ -163,49 +174,142 @@ func (r *ApplicationPullServiceAccountCreator) getImageRepositoryPullSecretNames
 	return pullSecretNames, nil
 }
 
-// createApplicationServiceAccount creates application service account with provided pull secrets
-func (r *ApplicationPullServiceAccountCreator) createApplicationServiceAccount(ctx context.Context, serviceAccountName string, application *appstudioredhatcomv1alpha1.Application, pullSecretNames []string) error {
-	log := ctrllog.FromContext(ctx).WithValues("serviceAccountName", serviceAccountName, "namespace", application.Namespace)
+// dockerConfigJson represents the structure of a .dockerconfigjson secret
+type dockerConfigJson struct {
```

Comment:
move it to start before that AggregatedPullSecretCreator struct

Reply:
Moved